core/corehttp/gateway_handler.go

```diff
@@ -125,6 +125,10 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	i.addUserHeaders(w) // ok, _now_ write user's headers.
 	w.Header().Set("X-IPFS-Path", urlPath)
 
+	//robots-header to prevent search-engines from indexing the gateway because of the duplicate-content problematic when makeing the gateway public
+	w.Header().Set("X-Robots-Tag", "noindex")
```

Comment:
but agree good to have in one place. so perhaps this roadmap:

- [ ] consider how these options play with #1751
- [ ] perhaps make an option like `Gateway.Robots.AllowIndexing`, default to false.
- [ ] option drives this header, and `/robots.txt`.
- [ ] sharness tests for the config option, expected header, and `/robots.txt`